### DISCUSSION TITLE
Error when adding a dot to the first segment in url

### DISCUSSION THREAD

**Discussion by ninnjak** [amannn/next-intl#464] [Q&A] [answered]
### Description

I'm facing the same issue outlined here: https://github.com/amannn/next-intl/issues/307

I have an application that uses the first segment for user profiles e.g. (http://localhost:3000/username) when the username contains a dot (http://localhost:3000/username.username) it attempts to pass username.username as the locale instead of the default locale. It works with all other locales when explicitly set in the url.

### Mandatory reproduction URL (CodeSandbox or GitHub repository)

https://codesandbox.io/p/sandbox/inspiring-sun-2fsrpw

### Reproduction description

Steps to reproduce:
1. Open reproduction
2. Navigate to /username.username
3. Returns 404 page

### Expected behaviour

The url should be resolved using the default locale

**Comment by amannn** [comment, marked as answer]
The reason why you're encountering this issue is that your middleware matcher excludes anything that contains a dot.

You can fix this by adjusting the matcher, but be careful that you're still excluding anything that is not a content route (e.g. favicon.ico, etc.). You could consider adding all static assets to a folder like `./public/assets` and then exclude that one from the middleware:

```tsx
export const config = {
  matcher: ["/((?!api|_next|assets).*)"],
};
```

Working demo: https://codesandbox.io/p/sandbox/youthful-haibt-xp92kh?file=/middleware.ts:15,1-17,3

I'll move this to a discussion since it's more a usage question than a bug report.
